django/contrib/admin/templatetags/admin_modify.py

```diff
@@ -43,14 +44,13 @@ def submit_row(context):
     """
     Displays the row of buttons for delete and save.
     """
-    opts = context['opts']
     change = context['change']
     is_popup = context['is_popup']
     save_as = context['save_as']
     show_save = context.get('show_save', True)
     show_save_and_continue = context.get('show_save_and_continue', True)
-    ctx = {
-        'opts': opts,
+    ctx = Context(context)
```

Comment:
The 3 tests that fail are:

- `test_save_as_new_with_inlines_with_validation_errors`
- `test_save_as_new_with_validation_errors`
- `test_save_as_new_with_validation_errors_with_inlines`

They all fail with: `KeyError: u'show_save_and_add_another'`.

I've tried to follow the execution order with `pdb` but it's very very hard to understand what is happening. Probably the safest choice is to create a new `Context` instance, the current implementation of `submit_row` (which seems to date back as far as to 2006!) creates a new dict, it does not modify the existing context, that probably happened for a reason, but I'm speculating.